python/tk_desktop/desktop_window.py

```diff
@@ -600,6 +605,64 @@ def _register_apps_tab(self):
     def contextMenuEvent(self, event):
         self.user_menu.exec_(event.globalPos())
 
+    def closeEvent(self, event):
+        # When closing the app programatically via self.close(),
+        # the event is not spontaenous and we know the state of self._is_quitting
```

Comment:
A small thing here, take it or leave it, but I had to look this up, because I wasn't sure if you were just calling the event spontaneous or if an event actually had a spontaneous state/parameter.
In the case of this I read that it is an event that was triggered from outside of the application.

Reply:
Yeah, I did a poor job here, I'll clarify the comments